definition/definition.go

```diff
@@ -192,42 +288,64 @@ type loopCfg struct {
 	}
 }
 
-func (l *loopCfg) Accumulated() bool {
+//DEPRECATED.
+func (l *oldloopCfg) Accumulated() bool {
 	return l.accumulate
 }
 
-func (l *loopCfg) DowhileCondition() expression.Expr {
+//DEPRECATED.
+func (l *oldloopCfg) DowhileCondition() expression.Expr {
 	return l.condition
 }
 
-func (l *loopCfg) GetIterate() interface{} {
+//DEPRECATED.
+func (l *oldloopCfg) GetIterate() interface{} {
 	return l.iterate
 }
 
-func (l *loopCfg) IterateEnabled() bool {
+//DEPRECATED.
+func (l *oldloopCfg) IterateEnabled() bool {
 	return l.iterate != nil
 }
 
-func (l *loopCfg) DowhileEnabled() bool {
+//DEPRECATED.
+func (l *oldloopCfg) DowhileEnabled() bool {
 	return l.condition != nil
 }
 
-func (l *loopCfg) Delay() int {
+//DEPRECATED.
+func (l *oldloopCfg) Delay() int {
 	return l.delay
 }
 
-func (l *loopCfg) RetryOnErrorEnabled() bool {
+//DEPRECATED.
+func (l *oldloopCfg) RetryOnErrorEnabled() bool {
 	return l.retryOnError.count > 0
 }
 
-func (l *loopCfg) RetryOnErrorCount() int {
+//DEPRECATED.
+func (l *oldloopCfg) RetryOnErrorCount() int {
 	return l.retryOnError.count
 }
 
-func (l *loopCfg) RetryOnErrorInterval() int {
+//DEPRECATED.
+func (l *oldloopCfg) RetryOnErrorInterval() int {
 	return l.retryOnError.interval
 }
 
+// Need To Remove
+type Loop interface {
+	Accumulated() bool
+	DowhileCondition() expression.Expr
+	GetIterate() interface{}
+	IterateEnabled() bool
+	DowhileEnabled() bool
+	Delay() int
+	RetryOnErrorEnabled() bool
```

Comment:
Why are these part of the loop interface?  The retry stuff is independent of loop